src/Encryption/Codec/EncryptionCodec.cs

```diff
@@ -14,26 +14,18 @@ public sealed class EncryptionCodec : IPayloadCodec, IDisposable
     private const int TagSize = 16;
     private static readonly ByteString EncodingByteString = ByteString.CopyFromUtf8("binary/encrypted");
 
+    private readonly byte[] key;
     private readonly ByteString keyIDByteString;
-    private readonly AesGcm aes;
 
-    public EncryptionCodec(string keyID = DefaultKeyID, byte[]? key = null)
+    public EncryptionCodec(string keyID = DefaultKeyID, byte[]? _key = DefaultKey)
```

Comment:
You can't set a default parameter to a non-constant value I don't believe (this is breaking CI). I think you should leave this as it is, and change below to `this.key = key ?? DefaultKey`.